### PR TITLE
Add an speedscope output format for profiling 

### PR DESCRIPTION
**Pull request by coolreader18**
I added some documentation to `README.md`, but essentially it now by default outputs a `flamescope.json` file, which you can upload to https://speedscope.app to view profiling information in a *much* nicer viewer than the html that flame outputs. I authored the `flamescope` crate, so if there are any problems that come up I can quickly fix them.

[Here's](https://www.speedscope.app/#profileURL=https://gist.githubusercontent.com/coolreader18/4d601fb73b9382a5c217f3ce9efcf849/raw/9b98b5814c110cd32e90fb9c3e73f3a598a0c9da/flamescope.json) a demo of speedscope with a RustPython profiling output.

### CODE REVIEW

**Comment by coolreader18** [comment]
You can also install speedscope from npm and view with `speedscope flamescope.json` as well.

File: README.md [hunk 1]
```diff
@@ -105,6 +105,23 @@ There also are some unit tests, you can run those with cargo:
 $ cargo test --all
 ```
 
+# Profiling
```
Comment (windelbouwman) [inline README.md]: Could you create a seperate document, maybe `development.md` with more in detail information for developers? I already moved the wasm part to a seperate document to keep the first readme file manageable.

File: src/main.rs [hunk 2]
```diff
@@ -55,15 +57,25 @@ fn run() -> Result<(), Box<std::error::Error>> {
...
-                 built with feature 'flame-it'",
-            ),
-    );
+    let app = app
+        .arg(
+            Arg::with_name("profile_output")
+                .long("profile-output")
+                .takes_value(true)
+                .help(
+                    "the file to output the profiling information to. present due to being \
```
Comment (windelbouwman) [inline src/main.rs]: I think you could skip the text "present due to being built with feature flame-it".

File: src/main.rs [hunk 3]
```diff
@@ -55,15 +57,25 @@ fn run() -> Result<(), Box<std::error::Error>> {
...
-            .long("profile-output")
-            .takes_value(true)
-            .help(
-                "the file to output the profile graph to. present due to being \
-                 built with feature 'flame-it'",
-            ),
-    );
+    let app = app
+        .arg(
+            Arg::with_name("profile_output")
```
Comment (windelbouwman) [inline src/main.rs]: Next to the profile output, could we have an option enabling / disabling tracing at runtime in addition to the feature at compile time?

File: src/main.rs [hunk 4]
```diff
@@ -92,13 +104,43 @@ fn run() -> Result<(), Box<std::error::Error>> {
     {
```
Comment (windelbouwman) [inline src/main.rs]: The contents of this block could be put into a seperate function which is appropriately named.

File: src/main.rs (hunk 3 above)
Comment (coolreader18) [reply src/main.rs]: I'm not sure, it'd require some sort of override of the `flame` dependency, and `flame-it` really shouldn't be enabled for a real release build, just developement. We could just disable outputting the data, I guess?

Comment (windelbouwman) [reply src/main.rs]: No, what I mean is that if the flame-it feature is enabled, you get flame trace options at the command line. One of these options is then wether or not you want to use tracing.

I fully agree that trace should be disabled in release build for end users. It's mostly for ourselves to detect slowness of rustpython.

**Comment by windelbouwman** [comment]
This speedscope thing is really something cool. I look forward to finding performance bottlenecks!

**Pull request merged**